src/dataflow/src/source/mod.rs

```diff
@@ -329,6 +329,10 @@ pub trait SourceInfo<Out> {
 /// Source-agnostic wrapper for messages. Each source must implement a
 /// conversion to Message.
 pub struct SourceMessage<Out> {
+    /// Which offset is the predecessor of this one. We know this information at the consumer
+    /// level, but our downstream components do not necessarily, so we tag it here to allow them to
+    /// reconstruct the total order.
+    pub predecessor: Option<MzOffset>,
```

Comment:
So I think we may need something like this in the future if we want to send messages from dataflow threads -> persister from a dataflow operator that is downstream of the consumer, but since we are not doing that now, I don't think we need to add this in to the SourceMessages that get sent downstream yet?